libraries/chain/controller.cpp

```diff
@@ -2812,17 +2836,24 @@ const producer_schedule_type&    controller::active_producers()const {
    return my->pending->get_pending_block_header_state().active_schedule;
 }
 
-const producer_schedule_type&    controller::pending_producers()const {
+const producer_authority_schedule& controller::pending_producers()const {
    if( !(my->pending) )
       return  my->head->pending_schedule.schedule;
 
    if( my->pending->_block_stage.contains<completed_block>() )
       return my->pending->_block_stage.get<completed_block>()._block_state->pending_schedule.schedule;
 
    if( my->pending->_block_stage.contains<assembled_block>() ) {
-      const auto& np = my->pending->_block_stage.get<assembled_block>()._unsigned_block->new_producers;
-      if( np )
-         return *np;
+      if (is_builtin_activated(builtin_protocol_feature_t::wtmsig_block_signatures)) {
+         auto exts = my->pending->_block_stage.get<assembled_block>()._unsigned_block->validate_and_extract_header_extensions();
+         if (exts.count(producer_schedule_change_extension::extension_id()))
+            return exts.lower_bound(producer_schedule_change_extension::extension_id())->second.get<producer_schedule_change_extension>();
```

Comment:
Dangling reference.

Reply:
Can't you return `*my->pending->_block_stage.get<assembled_block>()._new_producer_authority_cache` (assuming `has_value`) like you do below whether `WTMSIG_BLOCK_SIGNATURES` has activated or not? That would resolve the dangling reference issue and be more efficient.